Ports/qt6-qtbase/patches/0002-Disable-std-wchar-and-std-wstring.patch

```diff
@@ -0,0 +1,175 @@
+ typedef bool CollatorType;
+ const CollatorType NoCollator = false;
+ #endif
+diff --git a/src/corelib/text/qcollator_posix.cpp b/src/corelib/text/qcollator_posix.cpp
+index 92148fa3..215737c7 100644
+--- a/src/corelib/text/qcollator_posix.cpp
++++ b/src/corelib/text/qcollator_posix.cpp
+@@ -42,7 +42,6 @@
+ #include "qstring.h"
+ 
+ #include <cstring>
+-#include <cwchar>
+ 
+ QT_BEGIN_NAMESPACE
+ 
+@@ -65,7 +64,7 @@ void QCollatorPrivate::cleanup()
+ {
+ }
+ 
+-static void stringToWCharArray(QVarLengthArray<wchar_t> &ret, QStringView string)
++static void stringToWCharArray(QVarLengthArray<char> &ret, QStringView string)
+ {
+     ret.resize(string.length());
+     int len = string.toWCharArray(ret.data());
+@@ -85,10 +84,10 @@ int QCollator::compare(QStringView s1, QStringView s2) const
+     if (d->dirty)
+         d->init();
+ 
+-    QVarLengthArray<wchar_t> array1, array2;
++    QVarLengthArray<char> array1, array2;
+     stringToWCharArray(array1, s1);
+     stringToWCharArray(array2, s2);
+-    return std::wcscoll(array1.constData(), array2.constData());
++    return strcoll(array1.constData(), array2.constData());
+ }
+ 
+ QCollatorSortKey QCollator::sortKey(const QString &string) const
+@@ -96,13 +95,16 @@ QCollatorSortKey QCollator::sortKey(const QString &string) const
+     if (d->dirty)
+         d->init();
+ 
+-    QVarLengthArray<wchar_t> original;
++    QVarLengthArray<char> original;
+     stringToWCharArray(original, string);
+-    QList<wchar_t> result(original.size());
++    QList<char> result(original.size());
++    std::copy(original.cbegin(), original.cend(), result.begin());
++#if 0
+     if (d->isC()) {
+         std::copy(original.cbegin(), original.cend(), result.begin());
+     } else {
+-        size_t size = std::wcsxfrm(result.data(), original.constData(), string.size());
++        size_t size = strlen(original);//std::wcsxfrm(result.data(), original.constData(), string.size());
++        strncpy(result, original, size);
+         if (size > uint(result.size())) {
+             result.resize(size+1);
+             size = std::wcsxfrm(result.data(), original.constData(), string.size());
+@@ -110,12 +112,13 @@ QCollatorSortKey QCollator::sortKey(const QString &string) const
+         result.resize(size+1);
+         result[size] = 0;
+     }
++#endif
+     return QCollatorSortKey(new QCollatorSortKeyPrivate(std::move(result)));
+ }
+ 
+ int QCollatorSortKey::compare(const QCollatorSortKey &otherKey) const
+ {
+-    return std::wcscmp(d->m_key.constData(), otherKey.d->m_key.constData());
++    return strcmp(d->m_key.constData(), otherKey.d->m_key.constData());
+ }
+ 
+ QT_END_NAMESPACE
+diff --git a/src/corelib/text/qstring.h b/src/corelib/text/qstring.h
+index 219b0df2..373ceec5 100644
+--- a/src/corelib/text/qstring.h
++++ b/src/corelib/text/qstring.h
+@@ -57,6 +57,10 @@
+ #include <QtCore/qstringtokenizer.h>
+ 
+ #include <string>
++namespace std {
++    typedef string wstring;
++}
++
```

Comment:
I could look into working around this limitation for Clang later if getting a proper `wstring` support takes a long time. Otherwise, it'll be really easy to remove.

Reply:
This patch is no longer required, as PR #11324 has been merged, which added support for `std::wstring`. You will need to re-build the toolchain to make use of it.

Reply:
Nice! Definitely trying that out as soon as I'm close to a charger

Reply:
Yup, I can confirm this patch is no longer necessary. Thanks for letting me know. I'll remove it in the next force push (after QSerenityPlatform will have been merged to SerenityPorts)